### ISSUE TITLE
Named exports cause an error in istanbul-lib-instrument 1.9

### ISSUE THREAD

**Issue by mohicanlove** [istanbuljs/istanbuljs#110]
# Expected Behavior
Running jest with coverage enabled does not produce an error when using a named export

```javascript
export class App extends Component = {}
```

# Observed Behavior
Running jest with coverage enabled results in the error

```shell
src/App.js: We don't know what to do with this node type. We were previously a Statement but we can't fit in here?

      at NodePath.insertBefore (node_modules/babel-traverse/lib/path/modification.js:61:13)
      at VisitState.insertCounter (node_modules/istanbul-lib-instrument/dist/visitor.js:197:22)
      at VisitState.insertStatementCounter (node_modules/istanbul-lib-instrument/dist/visitor.js:224:18)
      at VisitState.coverStatement (node_modules/istanbul-lib-instrument/dist/visitor.js:340:10)
      at node_modules/istanbul-lib-instrument/dist/visitor.js:327:15
      at Array.forEach (native)
      at VisitState.wrappedEntry (node_modules/istanbul-lib-instrument/dist/visitor.js:326:15)
      at NodePath._call (node_modules/babel-traverse/lib/path/context.js:76:18)
      at NodePath.call (node_modules/babel-traverse/lib/path/context.js:48:17)
      at NodePath.visit (node_modules/babel-traverse/lib/path/context.js:105:12)

```

# Code (or Repository) that Reproduces Issue
https://github.com/mohicanlove/istanbul-coverage-example/

# Forensic Information
**Operating System:** OSX 10.13
**Environment Information**
https://gist.github.com/mohicanlove/64e8d5aae229e54fc724f66914a0cc64

**Comment by mohicanlove** [comment]
Digging in a little more. The error happens specifically when you combine a named export with a class definition

```javascript
export class App extends Component = {} //throws error
export const App = () => {} // works
```

**Comment by ljharb** [comment]
This is also breaking react-dates; see https://travis-ci.org/airbnb/react-dates/jobs/291060598

**Comment by chrisui** [comment]
@bcoe Would the fix here work for default exports as well? We have encountered an issue isolated to  istanbul (turning off jest coverage tracking and the tests pass) where we are exporting a default class extending the native `Error` object.

The error we're odd (not fully traced yet) but we're seeing the resultant error.constructor not match the newly extended class we expect.

We've encountered this when updating from `istanbul-lib-instrument@1.8.0`.
